src/app.js

```diff
@@ -0,0 +1,49 @@
+const express = require('express');
+const app = express();
+const static = express.static(__dirname + '/public');
+const configRoutes = require('./routes');
+const exphbs = require('express-handlebars');
+
+const handlebarsInstance = exphbs.create({
+  defaultLayout: 'main',
+  // Specify helpers which are only registered on this instance.
+  helpers: {
+    asJSON: (obj, spacing) => {
+      if (typeof spacing === 'number')
+        return new Handlebars.SafeString(JSON.stringify(obj, null, spacing));
+
+      return new Handlebars.SafeString(JSON.stringify(obj));
+    }
+  }
+});
+
+const rewriteUnsupportedBrowserMethods = (req, res, next) => {
+  // If the user posts to the server with a property called _method, rewrite the request's method
+  // To be that method; so if they post _method=PUT you can now allow browsers to POST to a route that gets
+  // rewritten in this middleware to a PUT route
+  if (req.body && req.body._method) {
+    req.method = req.body._method;
+    delete req.body._method;
+  }
+
+  // let the next middleware run:
+  next();
+};
+
+// public will be our route for static files
+app.use('/public', static);
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+app.use(rewriteUnsupportedBrowserMethods);
+
+// using express handlebars
+app.engine('handlebars', handlebarsInstance.engine);
+app.set('view engine', 'handlebars');
+
+configRoutes(app);
+
+app.listen(3000, () => {
+  console.log("We've now got a server!");
```

Comment:
might want to get rid of this print statement at some point and just keep the one that says the server is running. No need to change now.